src/main/java/org/glassfish/concurro/AsynchronousInterceptor.java

```diff
@@ -89,4 +97,58 @@ public Object intercept(InvocationContext context) throws Exception {
         });
         return resultFuture;
     }
+
+    private static <T> T lookupMES(Class<T> cls, String executor, String methodName) throws RejectedExecutionException {
+        T mes;
+        try {
+            Object lookupMes = new InitialContext().lookup(executor);
+            if (lookupMes == null) {
+                throw new RejectedExecutionException("ManagedExecutorService with jndi '" + executor + "' not found!");
+            }
+            if (!(cls.isInstance(lookupMes))) {
+                throw new RejectedExecutionException("ManagedExecutorService with jndi '" + executor + "' must be of type " + cls + ", found " + lookupMes.getClass().getName());
+            }
+            mes = (T) lookupMes;
+        } catch (NamingException ex) {
+            throw new RejectedExecutionException("ManagedExecutorService with jndi '" + executor + "' not found as requested by asynchronous method " + methodName);
+        }
+        return mes;
+    }
+
+    public CompletableFuture<Object> schedule(InvocationContext context, Method method, Asynchronous asynchAnnotation) {
+
+        // FIXME: challenge testScheduledAsynchWithInvalidJNDIName from TCK and then remove this useless block
+        // For runAt, executor is not used at all!
+        // check existence of executor
+        String executor = asynchAnnotation.executor();
+        executor = executor != null ? executor : "java:comp/DefaultManagedExecutorService"; // provide default value if there is none
+        lookupMES(ManagedExecutorService.class, executor, method.getName());
+
+        ManagedScheduledExecutorService mses = lookupMES(ManagedScheduledExecutorService.class, "java:comp/DefaultManagedScheduledExecutorService", method.getName());
+        CompletableFuture<Object> future = mses.newIncompleteFuture();
+        CompoundTrigger compoundTrigger = new CompoundTrigger(mses, context);
+
+        for (Schedule schedule : asynchAnnotation.runAt()) {
+            long skipIfLateBySeconds = schedule.skipIfLateBy();
+            ZoneId zone = schedule.zone().isEmpty() ? ZoneId.systemDefault() : ZoneId.of(schedule.zone());
+            CronTrigger trigger;
+            if (schedule.cron().isEmpty()) {
+                trigger = new CronTrigger(zone)
+                        .seconds(schedule.seconds())
+                        .minutes(schedule.minutes())
+                        .hours(schedule.hours())
+                        .daysOfWeek(schedule.daysOfWeek())
+                        .daysOfMonth(schedule.daysOfMonth())
```

Comment:
`CronTrigger.daysOfMonth` does not like empty array:
https://github.com/jakartaee/concurrency/blob/e75e79c161abfabcddacc960b7259c3586607f5c/api/src/main/java/jakarta/enterprise/concurrent/CronTrigger.java#L238-L242

> one or more day numbers ranging from 1 to 31

which results in:
- https://github.com/eclipse-ee4j/glassfish/issues/25297